pkg/network/usm/shared_libraries.go

```diff
@@ -249,27 +250,34 @@ func (w *soWatcher) Start() {
 			return nil
 		}
 
-		// report silently parsing /proc error as this could happen
-		// just exit processes
-		proc, err := process.NewProcess(int32(pid))
+		mapsPath := fmt.Sprintf("%s/%d/maps", w.procRoot, pid)
+		maps, err := os.Open(mapsPath)
 		if err != nil {
 			log.Debugf("process %d parsing failed %s", pid, err)
 			return nil
 		}
-		mmaps, err := proc.MemoryMaps(true)
-		if err != nil {
-			if log.ShouldLog(seelog.TraceLvl) {
-				log.Tracef("process %d maps parsing failed %s", pid, err)
-			}
-			return nil
-		}
-
-		root := fmt.Sprintf("%s/%d/root", w.procRoot, pid)
-		for _, m := range *mmaps {
-			for _, r := range w.rules {
-				if r.re.MatchString(m.Path) {
-					w.registry.register(root, m.Path, uint32(pid), r)
-					break
+		defer maps.Close()
+
+		scanner := bufio.NewScanner(bufio.NewReader(maps))
+
+		cache := make(map[string]struct{})
+		for scanner.Scan() {
+			line := scanner.Text()
+			cols := strings.Fields(line)
+			// ensuring we have at least 6 elements in the line and the path (6th column) starts with `/`
+			// (indicates a path, and not an anonymous path).
+			if len(cols) >= 6 && strings.HasPrefix(cols[5], "/") {
```

Comment:
may check is len(cols) > 6 we probably have `(deleted)` meaning we can't access to the `.so` later
`72d89db2f000-72d89dcc4000 r-xp 00000000 fd:04 12345 /pathtolib.so (deleted)`

Reply:
added a UT that tests the scenario https://github.com/DataDog/datadog-agent/pull/17946#discussion_r1250894965